src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -44,7 +44,7 @@
     private final TransactionControl txControl;
     private final EntityManager em;
 
-    private List<DeviceMemberEntity> validateDeviceMembers(Long parentId, Long vsId, Long memberId, String childName) {
+    private List<DeviceMemberEntity> getDeviceMembers(Long parentId, Long vsId, Long memberId, String childName) {
         CriteriaBuilder criteriaBuilder = IsmDeviceApi.this.em.getCriteriaBuilder();
         CriteriaQuery<DeviceMemberEntity> query = criteriaBuilder.createQuery(DeviceMemberEntity.class);
         Root<DeviceMemberEntity> r = query.from(DeviceMemberEntity.class);
```

Comment:
why would the parentid be null?

Reply:
we wanted to refactor common code and put this as a public method. This method is used both by the REST layer and also by the Plugin code.The REST layer bases the query on deviceId and Plugin code bases the query on VS id. This is distinguished in the common code based on deviceId or vsId. If deviceId is set it uses deviceId based query otherwise vsId.
This is how we (Emanoel and I) agree to refactor the common code.
Hope this clarifies.